js/app/components/player/player.tsx

```diff
@@ -1,63 +1,73 @@
-import { useRenditions, useSegment } from "@streamplace/components";
-import { usePlayerSelectedRendition } from "features/player/playerSlice";
-import { selectUserMuted } from "features/streamplace/streamplaceSlice";
+import {
+  getFirstPlayerID,
+  usePlayerStore,
+  useSegment,
+} from "@streamplace/components";
 import usePlatform from "hooks/usePlatform";
 import useStreamplaceNode from "hooks/useStreamplaceNode";
-import { uuidv7 } from "hooks/uuid";
-import { useEffect, useMemo, useState } from "react";
-import { useAppSelector } from "store/hooks";
+import { useEffect, useState } from "react";
 import { Text, View } from "tamagui";
-import Fullscreen from "./fullscreen";
+import { Fullscreen } from "./fullscreen";
 import {
-  IngestMediaSource,
   PlayerEvent,
   PlayerProps,
   PlayerStatus,
   PlayerStatusTracker,
 } from "./props";
 import PlayerProvider from "./provider";
 
-const HIDE_CONTROLS_AFTER = 2000;
 const OFFLINE_THRESHOLD = 10000;
 
-export function Player(props: Partial<PlayerProps>) {
+export function Player(
+  props: Partial<PlayerProps> & {
+    setFullscreen?: (fullscreen: boolean) => void;
+  },
+) {
   return (
     <PlayerProvider {...props}>
+      <PropUpFullscreen setFullscreen={props.setFullscreen} />
       <PlayerInner {...props} />
     </PlayerProvider>
   );
 }
```

Comment:
Doesn't need to be done now but when we make this an actual importable component we're gonna need to probably have a lot of `useEffect` for keeping the Zustand state in line with the user-provided props